### ISSUE TITLE
Request: Synchronize return values of UnitCastingInfo and UnitChannelInfo with Retail

### ISSUE THREAD

**Issue by Meorawr** [Stanzilla/WoWUIBugs#184]
During development of patch 2.5.1 and 1.14.0, the following API functions were re-added to these clients:

- [UnitCastingInfo](https://wowpedia.fandom.com/wiki/API_UnitCastingInfo)
- [UnitChannelInfo](https://wowpedia.fandom.com/wiki/API_UnitChannelInfo)

When added to the client, there was a change to these functions and their `notInterruptible` return value - it was completely removed. This then had the knock on effect of shifting the return values after it one-to-the-left, which just so happens to be one of the more important return values - the spell ID.

As a result, addons that need to query the spell ID of an active cast or channel spell need to do the following to account for the removed return value:

```lua
local name, text, texture, startTimeMS, endTimeMS, isTradeSkill, castID, notInterruptible, spellId;
if WOW_PROJECT_ID == WOW_PROJECT_MAINLINE then
    name, text, texture, startTimeMS, endTimeMS, isTradeSkill, castID, notInterruptible, spellId = UnitCastingInfo(unit);
else
    name, text, texture, startTimeMS, endTimeMS, isTradeSkill, castID, spellId = UnitCastingInfo(unit);
end
```

Generally we'd prefer there to be as few differences with APIs as possible between the clients. Each addon having to effectively guard its calls with the above is somewhat painful. As such, what we'd like is for the `notInterruptible` return value to be restored. Whether or not it actually returns `nil` all the time for "authenticity" isn't too much of a concern - we'd rather just have the signature be the same between the clients.

**Comment by Azilroka** [comment]
Additionally, It would be more beneficial if this was fully implemented into BCC / Classic Era so we can drop yet another chat library (LibCasterino) for cast bars / cast times.

**Comment by Meorawr** [comment]
I think this has been implemented in a future BCC patch (and possibly 1.14.1) based off some changes seen in the diff, but need to confirm.

**Comment by Meorawr** [comment]
![image](https://user-images.githubusercontent.com/287102/140650299-ab4b7eb0-419f-40b7-af16-e17d30ab455f.png)

UnitCastingInfo and CastingInfo now return the same number of values on both Classic Era (1.14.1) and Retail; it's _very_ likely this is also the case in a future BCC patch but will keep this open until it's confirmed. For what it's worth, the `notInterruptible` return value does seem to always be `nil` - but I've not tested this extensively.

**Comment by Meorawr** [comment]
For BCC this is resolved on the PTR as of patch 2.5.3.41402; all clients have 9 return values and `notInterruptible` appears to be `nil` (but, again, not tested extensively).
